app/models/project.rb

```diff
@@ -20,11 +20,14 @@ def config_for(metric)
 
   # Recompute latest scores and images WITHOUT refreshing raw data
   def update_scores_and_images
+    @current_scores ||= {}
+    @current_images ||= {}
     # Compute and cache newest scores and images based on latest sample data.
     # optional logic here to short-circuit the method if no new samples since last update
-    ProjectMetric.metric_names.each do |metric|
+    ProjectMetrics.metric_names.each do |metric|
       latest_sample = self.metric_samples.where(:metric_name => metric).first
-      metric_instance = class_for(metric).new({}, latest_sample.raw_data)
+      credentials_hash = self.configs.where(:metric_name => metric).first.options
```

Comment:
I think this can be DRY'd out and made a bit more elegant by using `Project#config_for` here, as in:
`credentials_hash = self.config_for(metric).options`, and maybe defining a corresponding method `#samples_for` (`latest_sample = self.samples_for(metric).limit(1)`, which also avoids fetching extra rows from DB if we really are only looking for most recent sample.